sippy-ng/src/component_readiness/ComponentReadiness.js

```diff
@@ -0,0 +1,861 @@
+import { useStyles } from '../App'
+import { useTheme } from '@material-ui/core/styles'
+import Button from '@material-ui/core/Button'
+import CheckBoxList from './CheckboxList'
+import ChevronLeftIcon from '@material-ui/icons/ChevronLeft'
+import ChevronRightIcon from '@material-ui/icons/ChevronRight'
+import clsx from 'clsx'
+import CompReadyRow from './CompReadyRow'
+import IconButton from '@material-ui/core/IconButton'
+import MenuIcon from '@material-ui/icons/Menu'
+import React from 'react'
+import ReleaseSelector from './ReleaseSelector'
+import Table from '@material-ui/core/Table'
+import TableBody from '@material-ui/core/TableBody'
+import TableCell from '@material-ui/core/TableCell'
+import TableHead from '@material-ui/core/TableHead'
+import TableRow from '@material-ui/core/TableRow'
+
+const dateFormat = 'yyyy-MM-dd HH:mm:ss'
+const days = 24 * 60 * 60 * 1000
+const initialTime = new Date()
+const initialStartTime = new Date(initialTime.getTime() - 30 * days)
+const initialEndTime = new Date(initialTime.getTime())
+const initialPrevStartTime = new Date(initialTime.getTime() - 30 * days)
+const initialPrevEndTime = new Date(initialTime.getTime())
+
+// Big query requests take a while so give the user the option to
+// abort in case they inadvertently requested a huge dataset.
+let abortController = new AbortController()
+const cancelFetch = () => {
+  abortController.abort()
+}
+
+// The API likes RFC3339 times and the date pickers don't.  So we use this
+// function to convert for when we call the API.
+// 4 digits, followed by a -, followed by 2 digits, and so on all wrapped in
+// a group so we can refer to them as $1 and $2 respectively.
+// We add a 'T' in the middle and a 'Z' on the end.
+function makeRFC3339Time(anUrlStr) {
+  const regex = /(\d{4}-\d{2}-\d{2})\s(\d{2}:\d{2}:\d{2})/g
+  const replaceStr = '$1T$2Z'
+  return anUrlStr.replace(regex, replaceStr)
+}
+
+// Given the data pulled from the API server, calculate an array
+// of columns using the first row.  Assumption: the number of columns
+// is the same across all rows.
+function getColumns(data) {
+  const row0Columns = data.rows[0].columns
+
+  let retVal = []
+  row0Columns.forEach((column) => {
+    let columnName = ''
+    for (const key in column) {
+      if (key !== 'status') {
+        columnName = columnName + ' ' + column[key]
+      }
+    }
+    retVal.push(columnName.trimStart())
+  })
+  return retVal
+}
+
+// This is used when the user clicks on one of the columns at the top of the table
+function singleRowReport(columnName) {
+  return '/componentreadiness/' + safeEncodeURIComponent(columnName) + '/tests'
+}
+
+export default function ComponentReadiness(props) {
+  console.log('ComponentReadiness start')
+  const classes = useStyles()
+  const theme = useTheme()
+
+  // Extract the url and get the parameters from it
+  const location = useLocation()
+  const groupByParameters = new URLSearchParams(location.search)
```

Comment:
Can you `useQueryParams` instead of the manual parsing? It has built-in types for handling array parameters. There's a lot of benefits to using the library. It's similar to my comments on Devan's PR: https://github.com/openshift/sippy/pull/940#discussion_r1174090862

There's some complexity in how you'll want to handle the parameters that I think using the library will make easier.

- If I load the component readiness page with URL parameters set, I think you don't want the user to press the "generate" report button, but rather just go and do it for them.  i.e., if I share the URL on Slack, I just want it to load directly
- If I adjust the parameters on the form, I don't want any of them to update the page until I click "Generate Report" -- the way to do that is going to be have to layers of variables, ones that the form components change, and then `Generate Report` will call the `useQueryParams` setters on the values in the forms. That in turn should trigger useEffect to call the API again automatically.

If this doesn't make sense we could talk more live